src/Compilers/Core/Portable/Operations/OperationExtensions.cs

```diff
@@ -10,6 +11,27 @@ namespace Microsoft.CodeAnalysis.Semantics
 {
     public static class OperationExtensions
     {
+        /// <summary>
+        /// This will check whether context around the operation has any error such as syntax or semantic error
+        /// </summary>
+        public static bool IsInvalid(this IOperation operation, Compilation compilation, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            // once we made sure every operation has Syntax, we will remove this condition
+            if (operation.Syntax == null)
+            {
+                return true;
+            }
+
+            var model = compilation.GetSemanticModel(operation.Syntax.SyntaxTree);
+            if (model == null)
+            {
+                // given operation is not part of the given compilation. return as invalid operation
+                return true;
```

Comment:
If the syntax tree is not part of the compilation, we should `throw ArgumentException()`. It looks like `Compilation.GetSemanticModel()` already does that, at least in C#.